src/aspire/source/__init__.py

```diff
@@ -341,29 +341,25 @@ def downsample(self, L):
         ensure(L <= self.L, "Max desired resolution should be less than the current resolution")
         logger.info(f'Setting max. resolution of source = {L}')
 
-        self.model_pipeline.downsample(resolution=L)
         self.generation_pipeline.add_xform(Downsample(resolution=L))
 
+        ds_factor = self.L / L
+        for f in set(self.filters):
+            f.scale(ds_factor)
+        self.offsets /= ds_factor
+
         self.L = L
         # Invalidate images
         self._im = None
 
-    def whiten(self, whiten_filter=None):
+    def whiten(self, whiten_filter):
         """
-        Modify the Source object in place by whitening + caching all images, and adding the appropriate whitening
-            filter to all available filters.
-        :param whiten_filter: Whitening filter to apply.
-        :return: On return, the Source object has been modified in place.
+        Modify the `ImageSource` in-place by appending a whitening filter to the generation pipeline.
+        :param whiten_filter: Whitening filter to apply, as a `Filter` object.
```

Comment:
So if the user is specifying the *whitening* filter, we do not need to take the negative square root before applying it. If, on the other hand, the user specifies the *noise PSD filter*, we do. So either the input is `noise_filter`, or we keep it as `whiten_filter` (but remove the `PowerFilter`  below).

Reply:
Great point. My bad - renamed argument to `noise_filter`